### ISSUE TITLE
finish writing unit tests TaskStatusChangeHandler

### ISSUE THREAD

**Issue by mgmccarthy** [HTBox/allReady#787]
@tonysurma please label with "jump-in". Part of Issue #620

**Comment by rmcc13** [comment]
If no one is working on this, I would like to pick it up.

**Comment by mgmccarthy** [comment]
@rmcc13. I'll try to get a to a code review of this tonight. Since I'm not a project owner, I don't have the ability to set Assignees, so if you're working on a PR, you can include a comment on this issue that you're currently working on it. Thanks

**Comment by rmcc13** [comment]
OK sounds good, I'm working on this.

**Comment by rmcc13** [comment]
Here is an overview of what I am working on. The last commit included unit tests for success verifying expected behavior. Each test made two assertions: 1) Make sure the `TaskSignup` object has the expected `TaskStatus` value and 2) make sure the handler's `Handle` method published a `TaskSignupStatusChanged` notification once, and only once.

#### Current tests from last commit:

- Setting `TaskStatus` to "Assigned"
- Transitioning `TaskStatus` from "Assigned" to "Accepted"
- Transitioning `TaskStatus` from "Assigned" to "Rejected"
- Transitioning `TaskStatus` from "Accepted" to "Completed"
- Transitioning `TaskStatus` from "Accepted" to "CanNotComplete"

#### Here is what I am planning to work on next:

###### Tests for success

- [ ] Transition `TaskStatus` from "CanNotComplete" to "Accepted"
- [ ] Transition `TaskStatus` from "Completed" to "Accepted"
- [ ] Transition `TaskStatus` from "Assigned" to "Completed"
- [ ] Transition `TaskStatus` from "Assigned" to "CanNotComplete"

###### Tests for failure

- [ ] Transitioning `TaskStatus` from "Accepted" or "Rejected" to "Accepted" should throw an exception
- [ ] Transitioning `TaskStatus` from "Accepted", "Rejected", "Completed" or "CanNotComplete" to "Rejected" should throw an exception
- [ ] Transitioning `TaskStatus` from "Rejected", "Completed" or "CanNotComplete" to "Completed" should throw an exception
- [ ] Transitioning "Rejected", "Completed", or "CanNotComplete" to "CanNotComplete" should throw an exception
- [ ] If the `Task` or `TaskSignup` objects do not exist, an exception should be thrown

*Note* These tests all enforce the current implementation, I'm not sure if all these tests are necessary, but I plan on working on these unless others say they aren't necessary.

**Comment by mgmccarthy** [comment]
@rmcc13, thanks for the update. I'll take a look at the code to make sure all the tests you're going to write need to be written.

I'll also try to get to that code review for you last commit on this Issue today at some point.

**Comment by rmcc13** [comment]
OK cool, thanks

**Comment by mgmccarthy** [comment]
@rmcc13, thanks for the contribution. This PR looks solid. @tonysurma, when you get a chance, go ahead and accept the PR

**Comment by tonysurma** [comment]
Probably missing something but where is the PR? Thx

**Comment by mgmccarthy** [comment]
@tonysurma, I see this commit above: https://github.com/rmcc13/allReady/commit/a1fe5b25bcef6b65099db61658ace74110ecc236, but I don't see a PR.

@rmcc13, did you mean to create a PR for this work, or are you waiting to create a PR when the rest of the tests are done?

If you're waiting, it's easier on the code reviewers/owners to do reviews of small PR's rather than large ones. Let us know.

**Comment by rmcc13** [comment]
I was waiting to create a PR, but I can do one now. I wasn't too sure of your workflow.

**Comment by mgmccarthy** [comment]
@rmcc13, you can create as many PR's as you want to against this issue. Makes the review process a little easier.

**Comment by rmcc13** [comment]
OK sounds good, thanks!

**Comment by tonysurma** [comment]
Thanks again @rmcc13.  I have invited you to the repo so I can assign this and other tasks to you etc.  Thanks!

**Comment by rmcc13** [comment]
@tonysurma @mgmccarthy Just wondering if you guys think anything else needs to be done or if I can pick up another issue.

**Comment by mgmccarthy** [comment]
@rmcc13, there were a bunch of jump-in issues added last night, mainly around handler testing. Feel free to pick any of them.  Thanks!

**Comment by tonysurma** [comment]
Those are good to look at as well as any P1-P3 issue in either of the June go live milestones in the issue list as those are the ones high priority for users.  Thanks!

**Comment by mgmccarthy** [comment]
@tonysurma these unit tests are done and this Issue can be closed
